ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -453,12 +454,14 @@ public void testDescribeStatement() throws Exception {
         new RunningQuery(
             queries.get(0).getStatementString(),
             ((PersistentQueryMetadata)queries.get(0)).getSinkNames(),
-            ((PersistentQueryMetadata)queries.get(0)).getQueryId().getId()));
+            new EntityQueryId(
+                ((PersistentQueryMetadata)queries.get(0)).getQueryId())));
     List<RunningQuery> readQueries = Collections.singletonList(
         new RunningQuery(
             queries.get(1).getStatementString(),
             ((PersistentQueryMetadata)queries.get(1)).getSinkNames(),
-            ((PersistentQueryMetadata)queries.get(1)).getQueryId().getId()));
+            new EntityQueryId(
+                ((PersistentQueryMetadata)queries.get(1)).getQueryId())));
```

Comment:
I think I know the answer, but just want to make sure: this tests the SerDe of the request/response in its entirety right? Ie. the request in `makeSingleRequest` is serialized to json, deserialized, processed, and then the same happens for the response? So `testDescription` encodes the full round trip Serde?

Reply:
No - this tests the resource KsqlResource class directly as an isolated unit. So there is no actual server and the entities themselves are not serialized/deserialized. This is a gap in general in our api testing. We should have explicit testing to make sure we aren't breaking compatibility. This pr takes one approach: https://github.com/confluentinc/ksql/pull/1362/files#diff-16ef026843e07afe37cbd8abf74bf36a. We should add something similar here.

Reply:
Compatibility is the wrong word here. That implies testing across versions. Which we should also have but seems harder. This is more just basic serialization/deserialization testing.

Reply:
ok. Let's add a full SerDe test. Maybe file an issue? and do a followup?